deps/rabbit/test/per_user_connection_channel_tracking_SUITE.erl

```diff
@@ -770,30 +771,40 @@ exists_in_tracked_connection_per_vhost_table(Config, VHost) ->
 exists_in_tracked_connection_per_vhost_table(Config, NodeIndex, VHost) ->
     exists_in_tracking_table(Config, NodeIndex,
         fun rabbit_connection_tracking:tracked_connection_per_vhost_table_name_for/1,
+        tracked_connection_per_vhost,
         VHost).
 
 exists_in_tracked_connection_per_user_table(Config, Username) ->
     exists_in_tracked_connection_per_user_table(Config, 0, Username).
 exists_in_tracked_connection_per_user_table(Config, NodeIndex, Username) ->
     exists_in_tracking_table(Config, NodeIndex,
         fun rabbit_connection_tracking:tracked_connection_per_user_table_name_for/1,
+        tracked_connection_per_user,
         Username).
 
 exists_in_tracked_channel_per_user_table(Config, Username) ->
     exists_in_tracked_channel_per_user_table(Config, 0, Username).
 exists_in_tracked_channel_per_user_table(Config, NodeIndex, Username) ->
     exists_in_tracking_table(Config, NodeIndex,
         fun rabbit_channel_tracking:tracked_channel_per_user_table_name_for/1,
+        tracked_channel_per_user,
         Username).
 
-exists_in_tracking_table(Config, NodeIndex, TableNameFun, Key) ->
+exists_in_tracking_table(Config, NodeIndex, TableNameFun, Table, Key) ->
     Node = rabbit_ct_broker_helpers:get_node_config(
                 Config, NodeIndex, nodename),
     Tab = TableNameFun(Node),
-    AllKeys = rabbit_ct_broker_helpers:rpc(Config, NodeIndex,
-                                           mnesia,
-                                           dirty_all_keys, [Tab]),
-    lists:member(Key, AllKeys).
+    try
+        AllKeys = rabbit_ct_broker_helpers:rpc(Config, NodeIndex,
+                                               mnesia,
+                                               dirty_all_keys, [Tab]),
+        lists:member(Key, AllKeys)
+    catch
+        _:_ ->
+            All = rabbit_ct_broker_helpers:rpc(Config, NodeIndex,
+                                               ets, lookup, [Table, Key]),
+            lists:keymember(Key, 1, All)
+    end.
```

Comment:
Could you please check the state of the feature flag instead of trying one then the other?

This will make sure that it's consistent.